chalk-solve/src/wf.rs

```diff
@@ -144,7 +144,8 @@ where
         Self { db, solver_choice }
     }
 
-    pub fn verify_struct_decl(&self, struct_id: StructId<I>) -> Result<(), WfError<I>> {
+    /// TODO: Currently only handles structs, may need more work for enums & unions
+    pub fn verify_adt_decl(&self, adt_id: AdtId<I>) -> Result<(), WfError<I>> {
```

Comment:
I'm not sure if this function generalizes to all ADTs in its current form, so I flagged it with this TODO.